apps/frontend/src/pages/EditListing.tsx

```diff
@@ -1,5 +1,453 @@
+  };
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = async (
+    submissionEvent,
+  ) => {
+    submissionEvent.preventDefault();
+
+    if (!priceError && !titleError && !sent) {
+      const successImages: boolean = await asyncListingImageWrapper();
+      const successLocation: boolean = await asyncListingLocationWrapper();
+
+      if (successImages && successLocation) {
+        _axios_instance
+          .patch("/listing/" + id, newListingObject)
+          .then(() => {
+            alert("Listing Updated!");
+            setSent(true);
+            navigate("/");
+          })
+          .catch(() => {
+            alert("Listing Update Failed");
+            setSent(false);
+          });
+      } else if (!successImages) {
+        alert("Images failed to upload, please try again later");
+        setSent(false);
+      } else {
+        alert(
+          "Error occurred when updating the listing, you may need to enable location permissions for this site",
+        );
+        setSent(false);
+      }
+    }
+  };
+
+  const updateListingTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    const title = event.target.value;
+    if (!title) {
+      setTitleError("This field is required");
+    } else {
+      setTitleError("");
+    }
+    updateNewListingPayload("title", title);
+  };
+
+  const updateListingDescription = (
+    event: ChangeEvent<HTMLTextAreaElement>,
+  ) => {
+    updateNewListingPayload("description", event.target.value);
+  };
+
+  const updateListingPrice = (event: ChangeEvent<HTMLInputElement>) => {
+    const priceValue = event.target.value;
+    console.log("PRICE VALUE:", priceValue);
+    const regex = /^\d+(.\d{1,2})?$/;
+    if (!regex.test(priceValue)) {
+      setPriceError(
+        "This price is not valid, please make sure the value is positive and in the form xx.xx",
+      );
+    } else {
+      setPriceError("");
+      updateNewListingPayload("price", priceValue ? parseFloat(priceValue) : 0);
+    }
+  };
+
+  const updateListingLocation = () => {
+    try {
+      const currentLocation: LocationObject = { latitude: 0, longitude: 0 };
+      navigator.geolocation.getCurrentPosition((position) => {
+        const currentLatitude = position.coords.latitude;
+        const currentLongitude = position.coords.longitude;
+        if (currentLatitude !== 0 && currentLongitude !== 0) {
+          currentLocation.latitude = currentLatitude;
+          currentLocation.longitude = currentLongitude;
+        }
+      });
+      updateNewListingPayload("location", currentLocation);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  };
+
+  const asyncListingLocationWrapper = async (): Promise<boolean> => {
+    try {
+      return updateListingLocation();
+    } catch (error) {
+      return false;
+    }
+  };
+
+  const isImageValid = (url: string) => {
+    try {
+      return url.startsWith("data:image/");
+    } catch (error) {
+      return false;
+    }
+  };
```

Comment:
```suggestion
  const isImageValid = (url: string) => {
      return url.startsWith("data:image/");
  };
```

This should never throw an error

Reply:
There was a file that I uploaded that did cause an error earlier when I was manually testing it, I can't remember which file but I'd feel more comfortable if we left this in. Especially since it shouldn't add basically any runtime affecting the lighthouse score.

Reply:
I'd guess that the error was thrown elsewhere and not from this exact function, but no harm in leaving it in so go ahead